integration/bundle/init_test.go

```diff
@@ -99,6 +116,144 @@ func TestBundleInitOnMlopsStacks(t *testing.T) {
 	assert.Contains(t, job.Settings.Name, fmt.Sprintf("dev-%s-batch-inference-job", projectName))
 }
 
+func TestBundleInitTelemetryForDefaultTemplates(t *testing.T) {
+	projectName := testutil.RandomName("name_")
+
+	tcases := []struct {
+		name         string
+		args         map[string]string
+		expectedArgs map[string]string
+	}{
+		{
+			name: "dbt-sql",
+			args: map[string]string{
+				"project_name":     fmt.Sprintf("dbt-sql-%s", projectName),
+				"http_path":        "/sql/1.0/warehouses/id",
+				"default_catalog":  "abcd",
+				"personal_schemas": "yes, use a schema based on the current user name during development",
+			},
+			expectedArgs: map[string]string{
+				"personal_schemas": "yes, use a schema based on the current user name during development",
+			},
+		},
+		{
+			name: "default-python",
+			args: map[string]string{
+				"project_name":     fmt.Sprintf("default_python_%s", projectName),
+				"include_notebook": "yes",
+				"include_dlt":      "yes",
+				"include_python":   "no",
+			},
+			expectedArgs: map[string]string{
+				"include_notebook": "yes",
+				"include_dlt":      "yes",
+				"include_python":   "no",
+			},
+		},
+		{
+			name: "default-sql",
+			args: map[string]string{
+				"project_name":     fmt.Sprintf("sql_project_%s", projectName),
+				"http_path":        "/sql/1.0/warehouses/id",
+				"default_catalog":  "abcd",
+				"personal_schemas": "yes, automatically use a schema based on the current user name during development",
+			},
+			expectedArgs: map[string]string{
+				"personal_schemas": "yes, automatically use a schema based on the current user name during development",
+			},
+		},
+	}
+
+	for _, tc := range tcases {
+		ctx, _ := acc.WorkspaceTest(t)
+
+		// Configure a telemetry logger in the context.
+		ctx = telemetry.ContextWithLogger(ctx)
+
+		tmpDir1 := t.TempDir()
+		tmpDir2 := t.TempDir()
+
+		// Create a config file with the project name and root dir
+		initConfig := tc.args
+		b, err := json.Marshal(initConfig)
+		require.NoError(t, err)
+		err = os.WriteFile(filepath.Join(tmpDir1, "config.json"), b, 0o644)
+		require.NoError(t, err)
+
+		// Run bundle init
+		assert.NoDirExists(t, filepath.Join(tmpDir2, tc.args["project_name"]))
+		testcli.RequireSuccessfulRun(t, ctx, "bundle", "init", tc.name, "--output-dir", tmpDir2, "--config-file", filepath.Join(tmpDir1, "config.json"))
+		assert.DirExists(t, filepath.Join(tmpDir2, tc.args["project_name"]))
+
+		// Assert the telemetry payload is correctly logged.
+		logs, err := telemetry.GetLogs(ctx)
+		require.NoError(t, err)
+		require.Equal(t, 1, len(logs))
+		event := logs[0].Entry.DatabricksCliLog.BundleInitEvent
+		assert.Equal(t, event.TemplateName, tc.name)
+		assert.Equal(t, event.TemplateEnumArgs, tc.expectedArgs)
+	}
+}
+
+func TestBundleInitTelemetryForCustomTemplates(t *testing.T) {
```

Comment:
If we send telemetry on bundle init error, can you add a test for this as well

Reply:
We don't this PR is only for successful `bundle init` invocations.